presto-parquet/src/main/java/io/prestosql/parquet/ParquetCompressionUtils.java

```diff
@@ -89,15 +89,26 @@ private static Slice decompressGzip(Slice input, int uncompressedSize)
             return EMPTY_SLICE;
         }
 
-        DynamicSliceOutput sliceOutput = new DynamicSliceOutput(uncompressedSize);
         byte[] buffer = new byte[uncompressedSize];
-        try (InputStream gzipInputStream = new GZIPInputStream(input.getInput(), GZIP_BUFFER_SIZE)) {
-            int bytesRead;
-            while ((bytesRead = gzipInputStream.read(buffer)) != -1) {
-                sliceOutput.write(buffer, 0, bytesRead);
+        int bytesRead = 0;
+        boolean eos = false;
+        try (GZIPInputStream gzipInputStream = new GZIPInputStream(input.getInput(), min(GZIP_BUFFER_SIZE, input.length()))) {
+            int n;
```

Comment:
I'd rename this variable to `read`, and maybe the other one to `totalBytesRead` to have a clear separation.

Reply:
Have you considered using Guava's `ByteStreams.read()` instead of rolling you own? (it may be tricky due to the error handling below, but worth considering)